daemon/daemon.go

```diff
@@ -81,6 +84,8 @@ func (c *Command) canAccess(r *http.Request, user *auth.UserState) bool {
 		}
 
 		isUser = true
+	} else if c.SnapOK {
```

Comment:
could you also always log the error (using `logger.Noticef`) if it's not `errNoUID`?